airbyte-integrations/connectors/source-twilio/source_twilio/streams.py

```diff
@@ -0,0 +1,316 @@
+#
+# Copyright (c) 2020 Airbyte
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+from abc import ABC, abstractmethod
+from typing import Any, Iterable, Mapping, MutableMapping, Optional
+from urllib.parse import parse_qsl, urlparse
+
+import pendulum
+import requests
+from airbyte_cdk.models import SyncMode
+from airbyte_cdk.sources.streams.http import HttpStream
+
+
+class TwilioStream(HttpStream, ABC):
+    url_base = "https://api.twilio.com/2010-04-01/"
+    primary_key = "sid"
+    page_size = 100
+
+    @property
+    def data_field(self):
+        return self.name
+
+    def path(self, **kwargs):
+        return f"{self.name.title()}.json"
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        stream_data = response.json()
+        next_page_uri = stream_data.get("next_page_uri")
+        if next_page_uri:
+            next_url = urlparse(next_page_uri)
+            next_page_params = dict(parse_qsl(next_url.query))
+            return next_page_params
+
+    def parse_response(self, response: requests.Response, stream_state: Mapping[str, Any], **kwargs) -> Iterable[Mapping]:
+        """
+        :return an iterable containing each record in the response
+        """
+        records = response.json().get(self.data_field, [])
+        yield from records
+
+    def request_params(self, next_page_token: Mapping[str, Any] = None, **kwargs) -> MutableMapping[str, Any]:
+        params = super().request_params(next_page_token=next_page_token, **kwargs)
+        params["PageSize"] = self.page_size
+        if next_page_token:
+            params.update(**next_page_token)
+        return params
+
+
+class IncrementalTwilioStream(TwilioStream, ABC):
+    cursor_field = "date_updated"
+
+    def __init__(self, start_date: str = None, **kwargs):
+        super().__init__(**kwargs)
+        self._start_date = start_date
+
+    @property
+    def incremental_filter_field(self) -> str:
```

Comment:
Why did you decide to use `property` in this case, because we do not plan to rename this variable in any way, and we do not make it abstract either?